src/parse.rs

```diff
@@ -258,6 +263,13 @@ impl<'a> Parser<'a> {
                 }
             }
             return Err(Error::InvalidBackref);
+        } else if b == b'k' {
+            if let Some((id, skip)) = parse_id(self.re, ix) {
+                if let Some(group) = self.named_groups.get(id) {
+                    return Ok((ix + skip, Expr::Backref(*group)));
+                }
+            }
+            return Err(Error::InvalidGroupName);
```

Comment:
Let's use a different error for this case, something like `InvalidGroupNameBackref`

Reply:
I agree. I'll work on this.